src/transformations/horner.cpp

```diff
@@ -343,40 +408,89 @@ polynomial evaluation engine.
     sz    =  2*transformation->order + 2;
     range =  transformation->range;
 
+    const bool iterative_inverse = direction == PJ_INV && transformation->has_only_fwd;
+
     if (direction==PJ_FWD) {                              /* forward */
-        cb =  transformation->fwd_c;
-        c  =  cb + sz;
         e  =  position.u - transformation->fwd_origin->u;
         n  =  position.v - transformation->fwd_origin->v;
         if (transformation->uneg)
             e  =  -e;
         if (transformation->vneg)
             n  =  -n;
     } else {                                              /* inverse */
-        cb =  transformation->inv_c;
-        c  =  cb + sz;
-        e  =  position.u - transformation->inv_origin->u;
-        n  =  position.v - transformation->inv_origin->v;
-        if (transformation->uneg)
-            e  =  -e;
-        if (transformation->vneg)
-            n  =  -n;
+        if (!iterative_inverse) {
```

Comment:
There's a number of these conditional blocks in the code. Originally this was due to the *Poder dual autochecking* approach, where forward and inverse cases were handled by the same code. Now, having 3 different cases singing in different voices in the same code, it begins to look a bit hairy.

Now, while you still have a good grasp of this code, you should consider following up with another pull request (once this one is merged), untangling the three cases into separate code paths/functions.